nifi-nar-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketServer.java

```diff
@@ -213,6 +214,14 @@ public void configure(WebSocketServletFactory webSocketServletFactory) {
             webSocketServletFactory.setCreator(this);
         }
 
+        @Override
+        public void init() throws ServletException {
+            // Set Component ClassLoader as Thread Context ClassLoader so that jetty-server classes are available to WebSocketServletFactory.Loader
+            final ClassLoader componentClassLoader = getClass().getClassLoader();
+            Thread.currentThread().setContextClassLoader(componentClassLoader);
+            super.init();
+        }
```

Comment:
The class loader for the servlet worker threads can be set on `ServletContextHandler` when you build up the Jetty server / servlets.
```
final ServletContextHandler contextHandler = new ServletContextHandler();
contextHandler.setClassLoader(Thread.currentThread().getContextClassLoader());
```
(I used the instance class loader here because it was originally in 1.12 but the component class loader can also be used).

I'm not sure which one is the right way to set the class loader but using the `ServletContextHandler` API seems to me less "hack".

Reply:
Thanks for the feedback @turcsanyip!  That makes sense, and provides a cleaner solution.  I confirmed that setting the `ServletContextLoader` `ClassLoader` works at runtime.